ql/src/java/org/apache/hadoop/hive/ql/parse/repl/CopyUtils.java

```diff
@@ -102,6 +103,12 @@ private void doCopyRetry(FileSystem sourceFs, List<ReplChangeManager.FileInfo> s
         doCopyOnce(sourceFs, pathList, destinationFs, destination, useRegularCopy);
       } catch (IOException e) {
         // If copy fails, fall through the retry logic
+        if (!(e instanceof FileNotFoundException)) {
+          LOG.info(" File operation failed with error : " + e.getMessage());
+          // looks like some network outrage, reset the file system object and retry.
+          sourceFs = pathList.get(0).getFileSystem(hiveConf);
+          destinationFs = destination.getFileSystem(hiveConf);
+        }
         isCopyError = true;
       }
       pathList = getFilesToRetry(sourceFs, srcFileList, destinationFs, destination, isCopyError);
```

Comment:
Need to catch IOException from getFilesToRetry as well as it may throw this exception for NN failures and need retry.

Reply:
In my opinion, if we cannot handle all IO failures, then this patch doesn't make sense.